### PR TITLE
README: better document the usage of hooks and feature files

### PR DESCRIPTION
**Pull request by marquiz**
Mention supported runtimes that the hooks can use and describe how hooks
and feature files can be used for injecting labels from other pods.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -336,7 +336,9 @@ feature sources.
 The *local* feature source gets its labels by two different ways:
 * It tries to execute files found under `/etc/kubernetes/node-feature-discovery/source.d/`
 directory. The hook files must be executable. When executed, the hooks are
-supposed to print all discovered features in `stdout`, one per line.
+supposed to print all discovered features in `stdout`, one per line. In
+addition to ELF binaries, runtimes currently supported by the NFD release image
```
Comment (Ethyling) [inline README.md]: Should we mention `statically linked ELF binaries` to prevent any problem with dynamic libraries?

Comment (marquiz) [reply README.md]: Thanks for thinking about this. I was also actually thinking about this, too, but wrote it in a hurry and did not mention this. But, now that you brought it up might be good to add a note. Possibly recommend statically linked binaries and mention that there is a very limited number of libraries available in the image (basically "try and see if it works")

Comment (marquiz) [reply README.md]: Pushed a new version of the PR. Static binaries are now mentioned as the recommended ELF format

**Pull request merged**